atom/browser/api/atom_api_extension.cc

```diff
@@ -220,6 +222,33 @@ void Extension::Enable(const std::string& extension_id) {
 }
 
 // static
+void Extension::Remove(const std::string& extension_id) {
+  const extensions::Extension* extension =
+    GetInstance()->extensions_.GetByID(extension_id);
+  if (extension) {
+    content::NotificationService::current()->Notify(
+      atom::NOTIFICATION_DISABLE_USER_EXTENSION_REQUEST,
+      content::Source<Extension>(GetInstance()),
+      content::Details<const extensions::Extension>(extension));
+  }
+
+  base::FilePath install_directory;
+  PathService::Get(component_updater::DIR_COMPONENT_USER, &install_directory);
+  extensions::file_util::UninstallExtension(install_directory, extension_id);
```

Comment:
this should run on the file thread and should probably also happen in `AtomExtensionSystem` to prevent uninstall of component extensions. See `ExtensionService::OnExtensionInstalled`